### DISCUSSION TITLE
Add Multithreading

### DISCUSSION THREAD

**Discussion by OmarAHex** [snobu/destreamer#388] [Ideas]
The program ran and downloaded the file with no errors, but the problem was the download speed (600 KB/s) was about 24x slower than my actual download speed (14MB/s). I suppose a multithreading or -nthreads option could help people download files faster, as usually with my own projects I need to use about 20 threads to saturate the transfer speed (up to about half of my actual).

PS. idk if this is something to post on discussion but I had no idea where to post suggestions

**Comment by grzglo** [comment]
Aria2c allows doing it with two different options, you can use parameter '-j' to define *number of simultaneous downloads* and  '-x' to set number of threads for each download. I believe this is the way new versions of destreamer are doing staff. In most cases, the bottleneck is CPU or GPU performance when decoding video to x265.
